source/exe/terminate_handler.cc

```diff
@@ -2,17 +2,38 @@
 
 #include <cstdlib>
 
+#include "envoy/common/exception.h"
+
 #include "source/common/common/logger.h"
 #include "source/server/backtrace.h"
 
+#include "absl/strings/str_format.h"
+
 namespace Envoy {
 
 std::terminate_handler TerminateHandler::logOnTerminate() const {
   return std::set_terminate([]() {
-    ENVOY_LOG(critical, "std::terminate called! (possible uncaught exception, see trace)");
+    logException(std::current_exception());
     BACKTRACE_LOG();
     std::abort();
   });
 }
 
+void TerminateHandler::logException(const std::exception_ptr current_exception) {
```

Comment:
nit: should current_exception be a const ref?

Reply:
What would we try to achieve with that? It's a pointer-like object, on most platforms of the same size as a pointer.  The convention appears to be to pass it by value.

Reply:
I think it's ok to leave as is, given that it's a pointer already

Reply:
got i; I saw "_ptr" and was thinking about Envoy's style of using that as a nickname for unique_ptr.